### PR TITLE
[SYCL] Generate property for default specialization constants values

### PR DESCRIPTION
**Pull request by maximdimakov**
Generate property not only for native mode helps to DPCPP RT get correct default values of specialization constants.
Signed-off-by: mdimakov <maxim.dimakov@intel.com>

### CODE REVIEW

**Review by AlexeySachkov — DISMISSED** [review]

**Review by s-kanaev — DISMISSED** [review]
The change looks good.
Though it should have a test.
I can't think of a good unit-test for RT here.
An E2E in intel/llvm-test-suite seems to be a good choice.

File: llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll [hunk 1]
```diff
@@ -235,7 +235,7 @@ attributes #3 = { nounwind }
 ; CHECK: !sycl.specialization-constants = !{![[#ID0:]], ![[#ID1:]], ![[#ID2:]], ![[#ID3:]], ![[#ID_COMPOS3:]], ![[#ID4:]], ![[#ID5:]]
 ;
 ; CHECK-DEF: !sycl.specialization-constants-default-values = !{![[#ID4:]], ![[#ID5:]], ![[#ID6:]], ![[#ID7:]], ![[#ID_COMPOS3_DEFAULT:]], ![[#ID8:]], ![[#ID9:]]
-; CHECK-RT-NOT: !sycl.specialization-constants-default-values
+; CHECK-RT: !sycl.specialization-constants-default-values
```
Comment (AlexeySachkov) [inline llvm/test/tools/sycl-post-link/spec-constants/SYCL-2020.ll:238]: Ideally, we should be unifying `CHECK-DEF` and `CHECK-RT` for that metadata into a common `CHECK`, i.e. we should check content of the metadata in both modes.

However, this test is quite complicated and messy already, I plan to refactor those LIT tests to make them cleaner and simpler. Therefore, I'm ok to accept this test as-is.

**Review by AlexeySachkov — COMMENTED** [review]
@maximdimakov, there seems to be a regression of `spec_constants_bool` ESIMD test, could you please take a look?

**Review by AlexeySachkov — APPROVED** [review]

**Comment by maximdimakov** [comment]
The problem with spec_const_bool.cpp in precommit happened due to XFAIL line in the test. Fix for it is in the :  https://github.com/intel/llvm-test-suite/pull/541

**Review by s-kanaev — APPROVED** [review]
LGTM

**Pull request merged**